src/templates/parseJavaTemplates.ts

```diff
@@ -0,0 +1,192 @@
+                if (setting) {
+                    userPromptedSettings.push(setting);
+                }
+            }
+        }
+    }
+
+    return {
+        isHttpTrigger: rawTemplate.id.toLowerCase().startsWith('httptrigger'),
+        id: rawTemplate.id,
+        name: getResourceValue(resources, rawTemplate.metadata.name),
+        defaultFunctionName: rawTemplate.metadata.defaultFunctionName,
+        language: ProjectLanguage.Java,
+        userPromptedSettings: userPromptedSettings,
+        categories: [TemplateCategory.Core]
+    };
+}
+
+function parseSetting(data: object, resources: IResources, variables: IVariables): IFunctionSetting {
+    const rawSetting: IRawSetting = <IRawSetting>data;
+    const enums: IEnumValue[] = [];
+    if (rawSetting.enum) {
+        for (const ev of rawSetting.enum) {
+            enums.push({
+                value: getVariableValue(resources, variables, ev.value),
+                displayName: getVariableValue(resources, variables, ev.display)
+            });
+        }
+    }
+
+    return {
+        name: getVariableValue(resources, variables, rawSetting.name),
+        resourceType: rawSetting.resource,
+        valueType: rawSetting.value,
+        defaultValue: rawSetting.defaultValue ? getVariableValue(resources, variables, rawSetting.defaultValue) : undefined,
+        label: getVariableValue(resources, variables, rawSetting.label),
+        enums: enums,
+        validateSetting: (value: string | undefined): string | undefined => {
+            if (rawSetting.validators) {
+                for (const validator of rawSetting.validators) {
+                    if (!value || value.match(validator.expression) === null) {
+                        return getVariableValue(resources, variables, validator.errorText);
+                    }
+                }
+            }
+
+            return undefined;
+        }
+    };
+}
+
+function getVariableValue(resources: IResources, variables: IVariables, data: string): string {
+    if (!isString(data)) {
+        // This evaluates to a non-string value in rare cases, in which case we just return the value as-is
+        return data;
+    }
+
+    const matches: RegExpMatchArray | null = data.match(/\[variables\(\'(.*)\'\)\]/);
+    data = matches !== null ? variables[matches[1]] : data;
+
+    return getResourceValue(resources, <string>data);
+}
+
+function getResourceValue(resources: IResources, data: string): string {
+    const matches: RegExpMatchArray | null = data.match(/\$(.*)/);
+    return matches !== null ? resources.en[matches[1]] : data;
+}
+
+export async function parseJavaTemplates(allTemplates: IFunctionTemplate[], functionAppPath: string, telemetryProperties?: TelemetryProperties): Promise<IFunctionTemplate[]> {
+    const javaScriptTemplates: IFunctionTemplate[] = allTemplates.filter((t: IFunctionTemplate) => t.language === ProjectLanguage.JavaScript);
```

Comment:
Please rename `basicJavaTemplates` to `backupJavaTemplates`. Also we shouldn't need to get  these most of the time.

I suggest something like this at the end of this function:
```typescript
if (templates.length > 0) {
    return templates;
} else {
    // move logic to get backup templates here
}
```